lib/util/error_code.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Error
+  # helps get information regarding current error codes.
+  class ConjurCode
+    def initialize(path)
+      @path = path
+      validate
+    end
+
+    def print_next_available
+      id = next_code_id
+      unless id
+        $stderr.puts "The path doesn't contain any files with Conjur error codes"
+        return
+      end
+      puts format("The next available error number is %d ( CONJ%05dE )", id, id)
+    end
+
+    # separate data-gathering from printing
+    def next_code_id
+      max_code = existing_codes.max
+      max_code ? max_code + 1 : nil
+    end
+
+    def existing_codes
+      codes = File.foreach(@path).map do |line|
+        match = /code: \"CONJ(?<num>[0-9]+)E\"/.match(line)
+        match ? match[:num].to_i : nil
+      end.compact
+      codes.compact
+    end
+
+    def validate
+      raise format("The following path:%s was not found", @path) unless File.file?(@path)
+    end
```

Comment:
To preserve margin:
```ruby
def validate
  unless File.file?(@path)
    raise format("The following path:%s was not found", @path)
  end
end
```
Generally, think: "If it fits easily on one line use the "unless/if at the end" syntax.  Once it overflows, just give up and go for traditional block syntax.